File: core/src/org/yeb/model/Edge.java
package org.yeb.model;

public class Edge {

    public final int id1;
    public final int id2;

    Edge(int id1, int id2) {
        this.id1 = id1;
        this.id2 = id2;
    }

    @Override
    public int hashCode() {
        return id1 * id2;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) {
            return true;
        }
        if (other instanceof Edge) {
            Edge that = (Edge) other;
            return (id1 == that.id1 && id2 == that.id2)
                           || (id1 == that.id2 && id2 == that.id1);
        }
        return false;
    }

    @Override
    public String toString() {
        return "Edge{" + id1 +", " + id2 + '}';
    }
}

File: core/src/org/yeb/Edge.kt
package org.yeb

class Edge(val id1: Int, val id2: Int) {

    override fun hashCode(): Int {
        return id1 * id2
    }

    override fun equals(other: Any?): Boolean {
        if (other is Edge) {
           return  (id1 == other.id1 && id2 == other.id2)
            || (id1 == other.id2 && id2 == other.id1)
        }
        return false
    }
}